modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -0,0 +1,513 @@
+    }
+    else
+      ring_radii_corr = _ring_radii;
+  }
+  auto mesh = buildReplicatedMesh(2);
+  // build the first slice of the polygon.
+  auto mesh0 = build_simple_slice(dynamic_pointer_cast<ReplicatedMesh>(mesh),
+                                  ring_radii_corr,
+                                  _ring_intervals,
+                                  _duct_sizes,
+                                  _duct_intervals,
+                                  _has_rings,
+                                  _has_ducts,
+                                  _pitch,
+                                  _num_sectors_per_side[0],
+                                  _background_intervals,
+                                  &_node_id_background_meta,
+                                  _num_sides,
+                                  1,
+                                  azimuthal_angles_array[0],
+                                  _block_id_shift,
+                                  _quad_center_elements,
+                                  _interface_boundary_id_shift);
+  // This loop builds add-on slices and stitches them to the first slice
+  for (unsigned int mesh_index = 1; mesh_index < _num_sides; mesh_index++)
+  {
+    auto mesh_tmp0 = buildReplicatedMesh(2);
+    auto mesh_tmp = build_simple_slice(dynamic_pointer_cast<ReplicatedMesh>(mesh_tmp0),
+                                       ring_radii_corr,
+                                       _ring_intervals,
+                                       _duct_sizes,
+                                       _duct_intervals,
+                                       _has_rings,
+                                       _has_ducts,
+                                       _pitch,
+                                       _num_sectors_per_side[mesh_index],
+                                       _background_intervals,
+                                       &_node_id_background_meta,
+                                       _num_sides,
+                                       mesh_index + 1,
+                                       azimuthal_angles_array[mesh_index],
+                                       _block_id_shift,
+                                       _quad_center_elements,
+                                       _interface_boundary_id_shift);
+
+    ReplicatedMesh other_mesh(*mesh_tmp);
+    MeshTools::Modification::rotate(other_mesh, 360.0 / _num_sides * mesh_index, 0, 0);
+    mesh0->prepare_for_use();
+    other_mesh.prepare_for_use();
+    mesh0->stitch_meshes(other_mesh, SLICE_BEGIN, SLICE_END, TOLERANCE, true);
+    other_mesh.clear();
+  }
+
+  // An extra step to stich the first and last slices together
+  mesh0->stitch_surfaces(SLICE_BEGIN, SLICE_END, TOLERANCE, true);
+
+  if (!_has_rings && !_has_ducts && _background_intervals == 1)
+    MooseMesh::changeBoundaryId(*mesh0, 1, OUTER_SIDESET_ID, false);
+
+  if (!_is_general_polygon)
+  {
+    auto mesh0_dup = mesh0->clone();
```

Comment:
why make a clone? The function takes a const

Reply:
It seems not working without a clone. The const is just a pointer.